ecommerce/extensions/payment/processors/braintree.py

```diff
@@ -216,9 +213,93 @@ def create_transaction(self, nonce, basket):
             raise GatewayError
 
         else:
-            # TODO Log the error!
-            # for error in result.errors.deep_errors:
-            #     print("attribute: " + error.attribute)
-            #     print("  code: " + error.code)
-            #     print("  message: " + error.message)
+            self.log_deep_errors(result.errors.deep_errors)
             raise GatewayError
+
+    def get_or_create_customer_payment_method(self, nonce, customer):
+        """
+        Retrieve a braintree customer payment method if it
+        does not exist create one
+
+        Arguments:
+            nonce - Braintree payment nonce
+            customer - Braintree Customer instance
+
+        Returns:
+            It will return braintree PaymentMethod instance or
+            throw GatewayError Exception
+        """
+        customer_id = "{}".format(customer.id)
+        try:
+            customer_payment_method = customer.payment_methods[0].token
+        except IndexError:
+            customer_payment_method = ''
+
+        try:
+            result = braintree.PaymentMethod.find(customer_payment_method)
+            token = result.token
+            logger.info('Found existing PaymentMethod [%s] for customer [%s].',
+                        token, customer_id)
+
```

Comment:
minor: this empty line makes it a bit confusing, like the `try` and `except` blocks are some separate blocks. At least to me :)